src/providers/mod.rs

```diff
@@ -17,10 +19,66 @@ mod mock;
 mod sendgrid;
 mod ses;
 mod smtp;
+#[cfg(test)]
+mod test;
 
 /// Email headers.
 pub type Headers = HashMap<String, String>;
 
+/// Build email MIME message.
+pub fn build_multipart_mime(
```

Comment:
Thanks! Let's make it private though, so the tests and the child modules can see it but the send module can't.